pkg/auth/providers/oidc/oidc_provider_test.go

```diff
@@ -427,6 +434,100 @@ func TestGetClaimInfoFromToken(t *testing.T) {
 	}
 }
 
+func TestGetGroupsFromClaimInfo(t *testing.T) {
+	type fields struct {
+		Name        string
+		Type        string
+		CTX         context.Context
+		AuthConfigs v3.AuthConfigInterface
+		Secrets     wcorev1.SecretController
+		UserMGR     user.Manager
+		TokenMGR    tokenManager
+	}
+	type args struct {
+		claimInfo ClaimInfo
+	}
+	type want struct {
+		groupNames []string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   want
+	}{
+		{
+			name: "get groups from claim info",
+			args: args{
+				claimInfo: ClaimInfo{
+					Groups: []string{"group1", "group2"},
+				},
+			},
+			want: want{
+				groupNames: []string{"group1", "group2"},
+			},
+		},
+		{
+			name: "roles and groups are combined",
+			args: args{
+				claimInfo: ClaimInfo{
+					Groups: []string{"group1", "group2"},
+					Roles:  []string{"role1", "role2"},
+				},
+			},
+			want: want{
+				groupNames: []string{"group1", "group2", "role1", "role2"},
+			},
+		},
+		{
+			name: "just roles",
+			args: args{
+				claimInfo: ClaimInfo{
+					Roles: []string{"role1", "role2"},
+				},
+			},
+			want: want{
+				groupNames: []string{"role1", "role2"},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			o := &OpenIDCProvider{
+				Name:        tt.fields.Name,
+				Type:        tt.fields.Type,
+				CTX:         tt.fields.CTX,
+				AuthConfigs: tt.fields.AuthConfigs,
+				Secrets:     tt.fields.Secrets,
+				UserMGR:     tt.fields.UserMGR,
+				TokenMGR:    tt.fields.TokenMGR,
+			}
+			got := o.getGroupsFromClaimInfo(tt.args.claimInfo)
+			// Assert the length of returned groups is as expected
+			if len(got) != len(tt.want.groupNames) {
+				t.Errorf("getGroupsFromClaimInfo() returned %d groups, want %d", len(got), len(tt.want.groupNames))
+			}
+
+			// Create a map for easier comparison
+			gotGroupNames := make(map[string]bool)
+			for _, principal := range got {
+				parts := strings.Split(principal.Name, "://")
+				if len(parts) == 2 {
+					gotGroupNames[parts[1]] = true
+				}
```

Comment:
This could probably be simplified by making `gotGroupNames` a slice of strings and appending `parts[1]` to it here.

Then you could use [slices.Equal](https://pkg.go.dev/slices#Equal) to compare `tt.want.groupNames` to `gotGroupNames`.

You could even remove the `len(got) == len(tt.want.groupNames) ` check as `slices.Equal` will fail if the slices are not equal?